src/test/crypto_tests.cpp

```diff
@@ -629,56 +616,27 @@ static void TestChaCha20Poly1305AEAD(bool must_succeed, unsigned int expected_aa
     BOOST_CHECK_EQUAL(expected_ciphertext_and_mac.size(), ciphertext_buf.size());
     BOOST_CHECK(memcmp(ciphertext_buf.data(), expected_ciphertext_and_mac.data(), ciphertext_buf.size()) == 0);
 
-    // manually construct the AAD keystream
-    cmp_ctx.SetIV(seqnr_aad);
-    cmp_ctx.Seek(0);
-    cmp_ctx.Keystream(cmp_ctx_buffer.data(), 64);
-    BOOST_CHECK(memcmp(expected_aad_keystream.data(), cmp_ctx_buffer.data(), expected_aad_keystream.size()) == 0);
-    // crypt the 3 length bytes and compare the length
-    uint32_t len_cmp = 0;
-    len_cmp = (ciphertext_buf[0] ^ cmp_ctx_buffer[aad_pos + 0]) |
-              (ciphertext_buf[1] ^ cmp_ctx_buffer[aad_pos + 1]) << 8 |
-              (ciphertext_buf[2] ^ cmp_ctx_buffer[aad_pos + 2]) << 16;
-    BOOST_CHECK_EQUAL(len_cmp, expected_aad_length);
-
-    // encrypt / decrypt 1000 packets
+    BOOST_CHECK(aead_in.DecryptLength(&out_len, ciphertext_buf.data()));
```

Comment:
This line implies that the decrypted length is len(AAD) + len(payload). I interpreted, [this](https://gist.github.com/jonasschnelli/c530ea8421b8d0e80c51486325587c52#v2-messages-structure) to mean that the decrypted length is the length of the ciphertext of the payload alone.

If this is intentional, can we make it clearer in the BIP? IIUC, typically, in other protocols, the length in the preamble is the length of the payload that follows. Did we want to do it that way?

Reply:
The BIP, the unit test and the bench test have been updated to be consistently clear that the encrypted length is `len(ciphertext_payload)` and not `len(ciphertext_payload + ciphertext_aad)`